Gems/Atom/RPI/Code/Include/Atom/RPI.Public/Pass/AttachmentReadback.h

```diff
@@ -43,14 +46,37 @@ namespace AZ
             };
 
             AttachmentReadback(const RHI::ScopeId& scopeId);
-
             virtual ~AttachmentReadback();
 
-            //! Readback a pass attachment.
+            //! Reads back a single pass attachment.
             //! @param readbackName is the name for the readback buffer. And it will be saved in ReadbackResult::m_name. If the name is empty, a name will be generated automatically.
             //! Return true if the pass attachment readback request was submitted.  
             //! The callback function set by SetCallback(CallbackFunction callback) will be called once the readback is finished
-            bool ReadPassAttachment(const PassAttachment* attachment, const AZ::Name& readbackName);
+            //! @param imageViewDescriptor If null, by default it is assumed that @attachment refers to Mip Level 0.
+            //!        When different than null, the image view descriptor can be used to specify which mip level is the attachment bound to.
+            bool ReadPassAttachment(const PassAttachment* attachment, const AZ::Name& readbackName, const RHI::ImageViewDescriptor* imageViewDescriptor = nullptr);
```

Comment:
RHI::ImageViewDescriptor is used to represent a group of continuous sub-resources. If you want to just read one sub-resource, it's better to use RHI::ImageSubresource

I think RHI::ImageViewDescriptor is useful if user want to get a group of continuous sub-resources with only need setup one struct for the purpose. And it can be useful for most of use cases which try to read all the mipmaps. 

But we may need to update the implementation so it can create multiple copy items rather than one. 

To support read multiple mipmaps which may not be continues, we can have another function use RHI::ImageSubresource as input. 

`bool ReadPassAttachment(const PassAttachment* attachment, const AZ::Name& readbackName, const RHI::ImageSubresource * imageSubresource= nullptr);`

Reply:
Maybe we should only have this one function 
`bool ReadPassAttachment(const PassAttachment* attachment, const AZ::Name& readbackName, const RHI::ImageViewDescriptor* imageViewDescriptor = nullptr)`

and remove the new `bool ReadPassAttachments(const AZStd::vector<ReadbackRequestInfo>& readbackAttachmentRequests);`function. 

This is because PassAttachment in a scope (ImageScopeAttachment) already contains ImageViewDescriptor information which makes sense if we want to use that when read the PassAttachment. 

If we only want to read one mip, we can construct the ImageViewDescriptor which only contains one mip.

Reply:
Great feedback, but I need a little bit of clarification... In my Compute Shader I have:
```
RWTexture3D m_mips[6]
```
I have several attachments "Mip0"..."Mip5", and I bind them independently where all of them are referring to the same Image, but different subresource, in other words one RHI::ImageViewDescriptor for each attachment:
```c++
AZ::RHI::ImageViewDescriptor viewDesc = AZ::RHI::ImageViewDescriptor::Create3D(m_texture3DAttachment->GetDescriptor().m_format,
                mipIndex, mipIndex, 0, pixelSizeForMip - 1);
binding->m_unifiedScopeDesc.SetAsImage(viewDesc);
``` 
So, the code snippet above is called for each Mip level, because each Mip is an attachment.  
So If I understand your point, it seems I could bind all Mips with a single attachment and assuming total mips = 6,
the code would look like this:
```
AZ::RHI::ImageViewDescriptor viewDesc = AZ::RHI::ImageViewDescriptor::Create3D(m_texture3DAttachment->GetDescriptor().m_format,
                0 /*top mip index*/, 5/*last mip index*/, 0, pixelSizeForMip - 1);
binding->m_unifiedScopeDesc.SetAsImage(viewDesc);
```

Reply:
@VickyAtAZ , per our conversation...
When **writing** to several Mip levels of the same UAV, we need to specify a unique ImageViewDescriptor for each mip level, and each mip level becomes its own attachment.
BUT when reading back, we can read back all Mips at once by referencing the attachment to Mip 0, and having  a single ImageViewDescriptor that describes the range of mip levels.  
  
Based on this clarification, I'm going to change the API, per your suggestion, to a single public function:  
```
bool ReadPassAttachment(const PassAttachment* attachment, const AZ::Name& readbackName, const RHI::ImageViewDescriptor* imageViewDescriptor = nullptr)
```
  
In the future if there's interest in being able to read back from different image attachments, the We'd have to modify `AZ::RPI::Pass` class to hold a list of `AttachmentReadback`s. But it is not necessary at the moment.